src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -1458,7 +1459,10 @@ private bool IsValidDataExtensionPropertyType(ITypeSymbol type)
                                 ParameterIndex = matchingConstructorParameter?.ParameterIndex ?? paramCount++,
                             };
 
-                            (propertyInitializers ??= new()).Add(propertyInitializer);
+                            if (propertyInitializers?.Any(p => propertyInitializer.Name.Equals(p.Name, GetStringComparison(options))) is not true)
```

Comment:
I suspect that the correct approach would be to deduplicate shadowed properties in the `ParsePropertyGenerationSpecs` method and not here.

Reply:
I started from there but I was lost with the fast-path flow, in particular around the `AddPropertyWithConflictResolution` function. Can you please help me by clarifying the link between fast-path properties and property initializers?

Reply:
> between fast-path properties and property initializers?

One governs generation of fast-path serialization methods, while the other controls what properties should be instantiated using member-initializer syntax in the deserializer. Both are fed data deriving from the disambiguating logic in `AddPropertyWithConflictResolution`.

Reply:
I have found a way to move the deduplication logic in the `ParsePropertyGenerationSpecs`. It looks way better like this.